File: tools/_shared/inc/models/gtfs_static/agency.py
import sqlite3

class Agency:
    def __init__(self, agency_id: str, agency_name, agency_url, agency_timezone, agency_lang, agency_phone):
        self.agency_id = agency_id
        self.agency_name = agency_name
        self.agency_url = agency_url
        self.agency_timezone = agency_timezone
        self.agency_lang = agency_lang
        self.agency_phone = agency_phone

    @staticmethod
    def init_from_db_row(db_row: sqlite3.Row):
        agency_id = db_row['agency_id']
        agency_name = db_row['agency_name']
        agency_url = db_row['agency_url']
        agency_timezone = db_row['agency_timezone']
        agency_lang = db_row['agency_lang']
        agency_phone = db_row['agency_phone']

        entry = Agency(agency_id, agency_name, agency_url, agency_timezone, agency_lang, agency_phone)

        return entry

File: apps/hrdf-check-duplicates/js/ts/models/gtfs/agency.ts
import { HRDF_Agency_DB } from "../../types/hrdf/agency_db"

export default class Agency {
    public agency_id: string
    public agency_name: string
    public agency_code: string | null
    public agency_url: string | null
    public agency_timezone: string | null
    public agency_lang: string | null
    public agency_phone: string | null

    constructor(agency_id: string, agency_name: string) {
        this.agency_id = agency_id
        this.agency_name = agency_name
        this.agency_code = null
        this.agency_url = null
        this.agency_timezone = null
        this.agency_lang = null
        this.agency_phone = null
    }

    public static initFromHRDFAgencyDB(agencyDB: HRDF_Agency_DB) {
        const agency_id = agencyDB.agency_id
        const agency_name = agencyDB.full_name_de
        
        const agency = new Agency(agency_id, agency_name)
        agency.agency_code = agencyDB.short_name

        return agency
    }
}